aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,443 @@ uid: fundamentals/servers/kestrel
+* `UseHttps(StoreName storeName, string subject, bool allowInvalid, StoreLocation location)`
+* `UseHttps(StoreName storeName, string subject, bool allowInvalid, StoreLocation location, Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(X509Certificate2 serverCertificate)`
+* `UseHttps(X509Certificate2 serverCertificate, Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(Action<HttpsConnectionAdapterOptions> configureOptions)`
+* `UseHttps(HttpsConnectionAdapterOptions httpsOptions)`
+
+`ListenOptions.UseHttps` parameters:
+
+* `filename` is the path and file name of a certificate file, relative to the directory that contains the app's content files.
+* `password` is the password required to access the X.509 certificate data.
+* `configureOptions` is an `Action` to configure the `HttpsConnectionAdapterOptions`. Returns the `ListenOptions`.
+* `storeName` is the certificate store from which to load the certificate.
+* `subject` is the subject name for the certificate.
+* `allowInvalid` indicates if invalid certificates should be considered, such as self-signed certificates.
+* `location` is the store location to load the certificate from.
+* `serverCertificate` is the X.509 certificate.
+* `httpsOptions` are options to configure HTTPS.
+
+In production, HTTPS must be explicitly configured. At a minimum, a default certificate must be provided.
+
+Supported configurations described next:
+
+* No configuration
+* Replace the default certificate from configuration
+* Change the defaults in code
+
+*No configuration*
+
+Kestrel listens on `http://localhost:5000` and `https://localhost:5001` (if a default cert is available).
+
+Specify URLs using the:
+
+* `ASPNETCORE_URLS` environment variable.
+* `--urls` command-line argument.
+* `urls` host configuration key.
+* `UseUrls` extension method.
+
+For more information, see [Server URLs](xref:fundamentals/hosting#server-urls) and [Overriding configuration](xref:fundamentals/hosting#overriding-configuration).
+
+The value provided using these approaches can be one or more HTTP and HTTPS endpoints (HTTPS if a default cert is available). Configure the value as a semicolon-separated list (for example, `"Urls": "http://localhost:8000;http://localhost:8001"`).
+
+*Replace the default certificate from configuration*
+
+Call `serverOptions.Configure(context.Configuration.GetSection("Kestrel"))` to specify the configuration section. A default HTTPS app settings configuration schema is available for Kestrel. Configure multiple endpoints, including the URLs and the certificates to use, either from a file on disk or from a certificate store.
+
+In the following *appSettings.json* example:
+
+* Set **AllowInvalid** to `true` to permit the use of invalid certificates (for example, self-signed certificates).
+* **HttpsDefaultCert** indicates use of the cert defined under **Certificates** > **Default** or the development certificate.
```

Comment:
This sounds like the name HttpsDefaultCert is special. Any https endpoint that does not specify a certificate will fall back to the the default specified the the Certificates section or elsewhere in code.